File: TP2/src/money.js
var money= (function(){
	function money(v,curr) {
	 if(v<0)
	    throw new MoneyNegative(v);
	 else if(curr.length>3)
	    throw new CurrencyBigger(curr);
	 else if(curr==null || v==null)
	 		throw new MoneyNull();
	 else if(!(typeof v === 'number'))
	 		throw new NotNumber(v);
	 else
		this.v=v;
		this.curr=curr;
	}

	money.prototype.getCurrency=function () {
		return this.curr;
	}
	money.prototype.getValue=function () {
		return this.v;
	}
	money.prototype.equals=function (otherM) {
	if(otherM.getValue()==this.getValue() && otherM.getCurrency().toUpperCase()==this.getCurrency().toUpperCase())
		return true ;
	if(otherM.getValue()==this.getValue() && otherM.getCurrency()==this.getCurrency())
	    return true;
	 else
	    return false;
	}
	money.prototype.toString=function toString() {
		return this.getValue()+" ("+this.getCurrency()+")" ;
	}
return money;
})();

File: tester/src/main/java/lille/Money.java
package lille;


public class Money {
	private int value;
	private String currency;
	

	Money(int value, String currency)
	{
		this.value=value;
		this.currency=currency;
	}

	public int getValue()
	{
		return this.value;
	}

	public String getCurrency()
	{
		return this.currency;
	}

    public String toString() {
     	return this.getValue()+" ("+this.getCurrency()+")";
    }
    
    @Override
    public boolean equals(Object object){
    	if(object==null)
    		return false;  	
    	if (this.getClass() != object.getClass())
    	    return false;
    	Money obje = (Money) object;
    	if((value==obje.getValue()) && (currency.equals(obje.getCurrency())))
    			return true;
    	if((value==obje.getValue()) && (currency.toUpperCase().equals(obje.getCurrency().toUpperCase())))
		        return true;
    	if((value!=obje.getValue()) || (!currency.equals(obje.getCurrency())))
    		    return false;
    	
	return true;
    
    	
    	
    }

}
